go/service/chat_boxer.go

```diff
@@ -169,3 +187,31 @@ func (b *chatBoxer) verify(data []byte, si chat1.SignatureInfo, prefix libkb.Sig
 	_, err := sigInfo.Verify()
 	return (err == nil)
 }
+
+// validSenderKey checks that the key is active for sender at ctime.
+func (b *chatBoxer) validSenderKey(sender gregor1.UID, key []byte, ctime gregor1.Time) (bool, error) {
+	kbSender, err := keybase1.UIDFromString(hex.EncodeToString(sender.Bytes()))
+	if err != nil {
+		return false, err
+	}
+	kid := keybase1.KIDFromSlice(key)
+	t := gregor1.FromTime(ctime)
+
+	user, err := libkb.LoadUser(libkb.NewLoadUserByUIDArg(b.G(), kbSender))
```

Comment:
And yeah, probably, we do want messages to be readable by me in `max,duder` if `duder` deleted his/her account.

I think on the other side of a reset or deletion, we need to probably invoke some server trust, since there's no way to reliably corroborate a deleted identity.

It could be we can handle both of those cases with the same codepath...